src/JoypadModule/src/Module.cpp

```diff
@@ -151,91 +158,130 @@ bool JoypadModule::close()
 bool JoypadModule::updateModule()
 {
     yarp::os::Bottle cmd, outcome;
-    float prepare{ 0.0 }, start{ 0.0 },
-          stop{ 0.0 }, pause{ 0.0 },
-          connectGoal{ 0.0 }, connectRpc{ 0.0 }, disconnect { 0.0 };
-
-    m_joypadController->getButton(m_prepareButton, prepare);
-    m_joypadController->getButton(m_startButton, start);
-    m_joypadController->getButton(m_stopButton, stop);
-    m_joypadController->getButton(m_pauseButton, pause);
-    m_joypadController->getButton(m_connectGoalButton, connectGoal);
-    m_joypadController->getButton(m_connectRpcButton, connectRpc);
-    m_joypadController->getButton(m_disconnectButton, disconnect);
+    float prepare{0.0}, start{0.0},
+        stop{0.0}, pause{0.0},
+        connectGoal{0.0}, connectRpc{0.0}, disconnect{0.0};
 
     // get the values from stick
-    double x{ 0.0 }, y{ 0.0 }, z{ 0.0 };
+    double linearVelocity{0.0}, lateralVelocity{0.0}, angularVeloctiy{0.0};
+    double x{0.0}, y{0.0}, z{0.0};
     m_joypadController->getAxis(0, y);
     m_joypadController->getAxis(1, x);
     m_joypadController->getAxis(2, z);
 
-    x = -deadzone(x);
-    y = -deadzone(y);
-    z = -deadzone(z);
+    angularVeloctiy = deadzone(z);
```

Comment:
I would suggest moving this inside the ``if (m_joypadType == "pedals")``, since for the other joypads, the angular velocity is obtained from ``y``

Reply:
done in [67700c6](https://github.com/robotology/walking-controllers/pull/188/commits/67700c6af013eeed2b4027b3cceeb6ac8fae1487)